liquibase-standard/src/main/java/liquibase/parser/AbstractFormattedChangeLogParser.java

```diff
@@ -610,6 +609,10 @@ protected InputStream openChangeLogFile(String physicalChangeLogLocation, Resour
         return resourceAccessor.getExisting(physicalChangeLogLocation).openInputStream();
     }
 
+    protected void handleInvalidEmptyPreconditionCase(ChangeLogParameters changeLogParameters, ChangeSet changeSet, Matcher preconditionMatcher) throws ChangeLogParseException {
+        throw new NotImplementedException();
```

Comment:
Please add a message explaining what we reached, like: 
```
throw new NotImplementedException("Invalid empty precondition found");
```

Reply:
Done